src/main/java/com/emc/pravega/perf/PerfStats.java

```diff
@@ -41,88 +39,107 @@
     private long bytes;
     private long maxLatency;
     private long totalLatency;
-    private long windowMaxLatency;
-    private long windowTotalLatency;
-    private long windowCount;
-    private long windowBytes;
-    final private long reportingInterval;
+    final private long windowInterval;
+    private timeWindow window;
```

Comment:
@GuardedBy("lock") ?

Reply:
Thanks Tom, But just personal choice, I have made all the public (Except one) as synchronized methods so that all the member variables stats are protected.